superdesk/io/feed_parsers/ninjs.py

```diff
@@ -182,16 +186,35 @@ def parse_renditions(self, renditions):
                 rend[rendition_name] = parsed_rendition
         return rend
 
-    def _format_qcodes(self, items):
+    def _format_qcodes(
+        self, items: List[Dict[str, Any]], language: Optional[str] = None, cv_name: Optional[str] = None
+    ) -> List[Dict[str, Any]]:
         subjects = []
         for item in items:
-            subject = {"name": item.get("name"), "qcode": item.get("code")}
+            if cv_name and not item.get("translations"):
+                cv_data = self.find_cv_item(cv_name, item.get("code"))
+                item = cv_data if cv_data else item
+
+            subject = {
```

Comment:
After looking more into this, I realised that setting the `name` attribute to the appropriate translated value happens on publish (in the [NINJSFormatter](https://github.com/superdesk/superdesk-core/blob/develop/superdesk/publish/formatters/ninjs_formatter.py#L77))

So what we should be doing here instead, is retrieving the `name`, `qcode`, `scheme` and `translations` attributes from the CV, falling back to the ingest source if not found.

Reply:
hi yep, you are right, but Canadians don't publish audio files from Superdesk but export them from another system, so maybe their translation value is missing.

Reply:
that's why I'm implementing this functionality here to retrieve the translation value from the CV if it is not found in the file.

Reply:
The `translations` field is not being populated on ingest. That is issue no. 1.
Issue 2. is we should use the CV entry first, and fallback to ingested value.

If a translation is missing, then it would be required in Superdesk's CV as well anyways

Reply:
And leave `name` as it is, and let the NINJS Formatter do the work (as it currently does with all other content types)